src/main/java/universe/UniverseInferenceAnnotatedTypeFactory.java

```diff
@@ -117,8 +120,7 @@ public void handleBinaryTree(AnnotatedTypeMirror atm, BinaryTree binaryTree) {
         }
     }
 
-    private static class UniverseInferencePropagationTreeAnnotater
-            extends PropagationTreeAnnotator {
+    private class UniverseInferencePropagationTreeAnnotater extends PropagationTreeAnnotator {
         public UniverseInferencePropagationTreeAnnotater(AnnotatedTypeFactory atypeFactory) {
```

Comment:
Is this a reference to the enclosing class? Or to the `realTypeFactory`?
By making the class non-static, there is now another type factory in scope and you access the qualifiers through that...